src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -307,6 +308,11 @@ private void LogCachingLogger(TContext rootContext, IAnalysisLogger logger, TCon
 
             if (results?.Count > 0)
             {
+                _run?.GetFileIndex(new ArtifactLocation { Uri = context.TargetUri },
+                                   addToFilesTableIfNotPresent: _persistArtifacts,
+                                   dataToInsert: _dataToInsert,
+                                   hashData: context.Hashes);
```

Comment:
we should only persist if:
1. we have results
2. if we have hashes/textfiles/binaryfiles flag enabled